src/Features/LanguageServer/Protocol/Features/Diagnostics/DefaultDiagnosticAnalyzerService.cs

```diff
@@ -1,283 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-using System;
-using System.Collections.Immutable;
-using System.Composition;
-using System.Diagnostics;
-using System.Linq;
-using System.Threading;
-using System.Threading.Tasks;
-using Microsoft.CodeAnalysis.Host.Mef;
-using Microsoft.CodeAnalysis.Options;
-using Microsoft.CodeAnalysis.PooledObjects;
-using Microsoft.CodeAnalysis.SolutionCrawler;
-using Roslyn.Utilities;
-
-namespace Microsoft.CodeAnalysis.Diagnostics
-{
-    [Shared]
-    [ExportIncrementalAnalyzerProvider(WellKnownSolutionCrawlerAnalyzers.Diagnostic, workspaceKinds: [])]
-    internal partial class DefaultDiagnosticAnalyzerService : IIncrementalAnalyzerProvider, IDiagnosticUpdateSource
```

Comment:
I did not expect this service to be deleted as part of this PR that is aimed at improving diagnostic mode flag and code conditioned on that flag. Is it possible that removing this default service implementation causes any issue, assuming this PR does not remove `WellKnownSolutionCrawlerAnalyzers.Diagnostic`? Should this deletion be done in a PR where we remove the solution crawler?

Reply:
> Should this deletion be done in a PR where we remove the solution crawler?

That's coming like 2-3 PRs from now :)